Assets/MixedRealityToolkit/_Core/EventDatum/Input/SourcePositionEventData.cs

```diff
@@ -37,46 +44,57 @@ public class SourcePositionEventData : SourceStateEventData
         /// <inheritdoc />
         public SourcePositionEventData(EventSystem eventSystem) : base(eventSystem) { }
 
+        public void Initialize(IMixedRealityInputSource inputSource, IMixedRealityController controller, TrackingState trackingState)
+        {
+            Initialize(inputSource, controller);
+            TrackingState = trackingState;
+        }
+
         /// <summary>
         /// Populates the event with data.
         /// </summary>
         /// <param name="inputSource"></param>
-        public void Initialize(IMixedRealityInputSource inputSource, Vector2 position)
+        /// <param name="controller"></param>
+        /// <param name="position"></param>
+        public void Initialize(IMixedRealityInputSource inputSource, IMixedRealityController controller, Vector2 position)
         {
-            Initialize(inputSource);
+            Initialize(inputSource, controller);
             TwoDofPosition = position;
         }
 
         /// <summary>
         /// Populates the event with data.
         /// </summary>
         /// <param name="inputSource"></param>
+        /// <param name="controller"></param>
         /// <param name="position"></param>
-        public void Initialize(IMixedRealityInputSource inputSource, Vector3 position)
+        public void Initialize(IMixedRealityInputSource inputSource, IMixedRealityController controller, Vector3 position)
         {
-            Initialize(inputSource);
+            Initialize(inputSource, controller);
             ThreeDofPosition = position;
         }
 
         /// <summary>
         /// Populates the event with data.
         /// </summary>
         /// <param name="inputSource"></param>
+        /// <param name="controller"></param>
         /// <param name="rotation"></param>
-        public void Initialize(IMixedRealityInputSource inputSource, Quaternion rotation)
+        public void Initialize(IMixedRealityInputSource inputSource, IMixedRealityController controller, Quaternion rotation)
         {
-            Initialize(inputSource);
+            Initialize(inputSource, controller);
             ThreeDofRotation = rotation;
         }
 
         /// <summary>
         /// Populates the event with data.
         /// </summary>
         /// <param name="inputSource"></param>
+        /// <param name="controller"></param>
         /// <param name="position"></param>
-        public void Initialize(IMixedRealityInputSource inputSource, SixDof position)
+        public void Initialize(IMixedRealityInputSource inputSource, IMixedRealityController controller, SixDof position)
         {
-            Initialize(inputSource);
+            Initialize(inputSource, controller);
             SixDofPosition = position;
```

Comment:
`SixDofPosition` seems misleadingly named. Also, does it give us anything that querying `ThreeDofPosition` and `ThreeDofRotation` (perhaps in addition to TrackingState) wouldn't?

Reply:
How is the name misleading?

The next name I thought of was Pose, but I thought people wouldn't understand right away.

Reply:
> . Also, does it give us anything that querying ThreeDofPosition and ThreeDofRotation (perhaps in addition to TrackingState) wouldn't?

I don't understand the question. Nothing here is being "queried", as it were. We're simply raising an event that the data in question was changed, for all the registered handler's to utilize.

Reply:
Just the "position" part of it, to me, since it's position + rotation. If that's commonly referred to as "six dof position", then we're good. I feel like I've normally heard it referred to having "six degrees of freedom", with the inference being position + rotation.

Reply:
>I feel like I've normally heard it referred to having "six degrees of freedom", with the inference being position + rotation.

Agreed, but Pose is also often used as a form of both Position & Rotation without scale.  Hmm

Reply:
> We're simply raising an event that the data in question was changed

Right, but do we get anything out of raising an event with a SixDof that we wouldn't get from raising an event with the two ThreeDofs? Seems like we're tracking duplicate data.

Reply:
> Pose is also often used as a form of both Position & Rotation without scale

Ooh, I personally like using pose

Reply:
I'm just afraid that most people won't know what it is right away.

Reply:
>Right, but do we get anything out of raising an event with a SixDof that we wouldn't get from raising an event with the two ThreeDofs? Seems like we're tracking duplicate data.

Better to raise a single event instead of two.

Reply:
>Seems like we're tracking duplicate data.

Not sure I understand.  The 3dof data isn't being set when we raise a sixdof event.

Reply:
> Not sure I understand

Possibly me neither! I'll take a better look tomorrow

Reply:
I recomment using Transform as that is what SixDof is, the position AND rotation.

Reply:
Though it is lacking scale that is commonly in a transform, so maybe pose is the best name :)